### PR TITLE
Fix : use any variable as unit, not just `ASR::Var` + support different kinds integers to be used as unit arg.

### PR DESCRIPTION
**Pull request by assem2002**
fixes https://github.com/lfortran/lfortran/issues/5728

### CODE REVIEW

**Comment by assem2002** [comment]
The addressed issue is resolved only with this commit https://github.com/lfortran/lfortran/pull/5803/commits/0d8f729d9227e2b9626a6f77962045f568fa6e30. I just noticed that we can't support other integer kinds but 4 as unit, so the subsequent commits fix that.

File: src/runtime/custom/lfortran_intrinsic_custom.f90 [hunk 1]
```diff
@@ -1,24 +1,55 @@
...
         integer, parameter :: LUN_MIN=0, LUN_MAX=1000
         logical :: opened
-        integer :: lun
-        integer, intent(out) :: unit
-
+        integer(4) :: lun ! no need for kind 8, we have limit of 1000 for now.
         do lun=LUN_MIN,LUN_MAX
             inquire(unit=lun,opened=opened)
             if (.not. opened) then
-                unit = lun
```
Comment (certik) [inline src/runtime/custom/lfortran_intrinsic_custom.f90:16]: This is problematic: it is using a loop variable as a return value. I would do the original, `unit = lun`, that seems clearer. Loop variables should not be used outside of the loop, and the above usage seems confusing to me.

**Review by certik — APPROVED** [review]
I think this is good. Left a minor comment that you can fix in a subsequent PR.

File: src/runtime/custom/lfortran_intrinsic_custom.f90 (hunk 1 above)
Comment (assem2002) [reply src/runtime/custom/lfortran_intrinsic_custom.f90]: done here https://github.com/lfortran/lfortran/pull/5803/commits/a1f7b3a839defa4eec8df756f1bc1baa302e5e1b

**Review by certik — APPROVED** [review]

**Pull request merged**